mesonpy/__init__.py

```diff
@@ -758,6 +758,11 @@ def __init__(
             if not value:
                 self._limited_api = False
 
+            if self._limited_api:
+                if bool(sysconfig.get_config_var("Py_GIL_DISABLED")):
+                    raise BuildError('Free-threaded CPython does not support '
+                                     'the Python limited API')
```

Comment:
You can most likely write the error message on one line without exceeding the maximum line length we enforce.

However, I'm not sure whether this error message is clear enough for users. Let's imagine an user trying to `pip install` a package from PyPI. If the user is using a nogil Python and the package uses the limited API, this message may not be enough to understand what is going on, ie why is meson-python trying to use the limited API. What do you think?

Maybe something on the lines of "The package requires the Python limited API but it is not supported by free-threaded Python" but with a better grammar.

By the way, I find that "free threaded" is not the most user-friendly name that could have been used for this. But this boat has sailed...

Reply:
I made it more verbose - along the lines of your suggestion, and also added a hint about how to get around it, which should always work.

> By the way, I find that "free threaded" is not the most user-friendly name that could have been used for this. But this boat has sailed...

Yeah agreed ... but unlikely to change at this point.